### PR TITLE
[DST-432]: integrate routing

### PR DESCRIPTION
**Pull request by aromko**
Primary I integrated the tanstack router library. So everything is setup to use it. I added a SidebarNavigation but good styles are missing.

### CODE REVIEW

File: src/routes/index.lazy.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { createLazyFileRoute } from '@tanstack/react-router';
+import App from '../App';
```
Comment (sarahgm) [inline src/routes/index.lazy.tsx:2]: why does it need to be called index.lazy.tsx?

File: src/routes/__root.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        }))
+      );
+
+export const Route = createRootRoute({
+  component: () => (
+    <>
+      <SideNavigation />
+      <div className="bg-gray-50">
+        <Outlet />
+        <Suspense>
```
Comment (sarahgm) [inline src/routes/__root.tsx:20]: doesn't it need a `fallback` ? @aromko

File: vite.config.ts [hunk 3]
```diff
@@ -19,20 +20,22 @@ export default defineConfig(async () => {
...
+      TanStackRouterVite(),
     ],
     test: {
       globals: true,
       environment: 'jsdom',
       setupFiles: './tests/setup.js',
       coverage: {
         exclude: [
-          '**/{*.d.ts,index.ts}',
+          '**/{*.d.ts,index.ts,*.gen.ts}',
```
Comment (sarahgm) [inline vite.config.ts:31]: what files are they?

Comment (aromko) [reply vite.config.ts:31]: Tanstack router create a routeTree.gen.ts. In this file are basically all routes listed which are generated from the routes directory. This file is automatically build when starting the dev server or build the production build, so we don't need to commit it.

File: src/routes/__root.tsx (hunk 2 above)
Comment (aromko) [reply src/routes/__root.tsx:20]: Hm. I just copied it. According to the docs the default is fallback={null} so I think it's ok. It's just for the dev tools.

Comment (sarahgm) [reply src/routes/__root.tsx:20]: ah okay understand

**Review by sarahgm — APPROVED** [review]

**Pull request merged**